api/src/main/java/org/openmrs/validator/VisitValidator.java

```diff
@@ -159,4 +167,29 @@ private void validateStopDatetime(Visit visit, Visit otherVisit, Errors errors)
 		}
 		
 	}
+	
+	private void validateVisitStartedBeforePatientBirthdate(Visit visit, Errors errors) {
+		if (visit.getPatient() == null || visit.getPatient().getBirthdate() == null || visit.getStartDatetime() == null) {
+			return;
+		}
+		
+		if (visit.getStartDatetime().before(getPatientBirthdateAdjustedIfEstimated(visit.getPatient()))) {
+			errors.rejectValue("startDatetime", "Visit.startDateCannotFallBeforeTheBirthDateOfTheSamePatient",
+			    "This visit has a start date that falls before the birthdate of the same patient.");
+		}
+	}
+	
+	private Date getPatientBirthdateAdjustedIfEstimated(Patient patient) {
+		Date birthday = patient.getBirthdate();
+		
+		if (patient.getBirthdateEstimated()) {
+			Calendar cal = Calendar.getInstance();
+			cal.setTime(birthday);
+			cal.add(Calendar.YEAR, Math.min(ESTIMATED_BIRTHDATE_ERROR_MARGIN_MINIMUM,
+			        new Double(Math.ceil(patient.getAge() * ESTIMATED_BIRTHDATE_ERROR_MARGIN)).intValue()));
```

Comment:
@djazayeri are you ok with `Math.ceil` (than for ex. Math.floor) here? This means we are more generous when computing the grace period